plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -62,6 +67,18 @@ public interface PredicatePushdownController
         return new DomainPushdownResult(simplifiedDomain, domain);
     };
 
+    static PredicatePushdownController pushdownDiscreteValues(Type type)
+    {
+        return (session, domain) -> {
+            Optional<Collection<Object>> expandedRange = domain.getValues().tryExpandRanges(getDomainCompactionThreshold(session));
+            if (expandedRange.isPresent() && expandedRange.get().size() > 1) {
+                Domain convertedDiscreteDomain = Domain.multipleValues(type, expandedRange.get().stream().collect(toImmutableList()));
```

Comment:
`Domain.create(ValueSet.copyOf(type, expandedRange.get()), domain.isNullAllowed())`

Reply:
thanks Raunaq, This helps not requiring additional condition in if clause.